dspace-spring-rest/src/test/java/org/dspace/app/rest/EntityTypeRestRepositoryIT.java

```diff
@@ -109,4 +52,42 @@ public void getAllEntityTypeEndpoint() throws Exception {
                            .matchEntityTypeEntry(entityTypeService.findByEntityType(context, "JournalIssue"))
                    )));
     }
+
+    @Test
+    public void getAllEntityTypeEndpointWithPaging() throws Exception {
+        //When we call this facets endpoint
+        getClient().perform(get("/api/core/entitytypes").param("size", "5"))
+
+                   //We expect a 200 OK status
+                   .andExpect(status().isOk())
+                   //The type has to be 'discover'
+                   .andExpect(jsonPath("$.page.size", is(5)))
+                   .andExpect(jsonPath("$.page.totalElements", is(7)))
+                   .andExpect(jsonPath("$.page.totalPages", is(2)))
+                   //There needs to be a self link to this endpoint
+                   .andExpect(jsonPath("$._links.self.href", containsString("api/core/entitytypes")))
+                   //We have 4 facets in the default configuration, they need to all be present in the embedded section
+                   .andExpect(jsonPath("$._embedded.entitytypes", containsInAnyOrder(
+                       EntityTypeMatcher
+                           .matchEntityTypeEntry(entityTypeService.findByEntityType(context, "Publication")),
+                       EntityTypeMatcher.matchEntityTypeEntry(entityTypeService.findByEntityType(context, "Person")),
+                       EntityTypeMatcher.matchEntityTypeEntry(entityTypeService.findByEntityType(context, "Project")),
+                       EntityTypeMatcher.matchEntityTypeEntry(entityTypeService.findByEntityType(context, "OrgUnit")),
+                       EntityTypeMatcher.matchEntityTypeEntry(entityTypeService.findByEntityType(context, "Journal"))
+                   )));
```

Comment:
you need to check that the second page works as well to verify the pagination

Reply:
The second page is verified now as well